server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowClusterTask.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.iotdb.rpc.TSStatusCode;
+import org.apache.iotdb.tsfile.read.common.block.TsBlockBuilder;
+import org.apache.iotdb.tsfile.utils.Binary;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+public class ShowClusterTask implements IConfigTask {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ShowClusterTask.class);
+
+  private static final IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+
+  private ShowClusterStatement showClusterStatement;
+
+  public ShowClusterTask(ShowClusterStatement showClusterStatement) {
+    this.showClusterStatement = showClusterStatement;
+  }
+
+  @Override
+  public ListenableFuture<ConfigTaskResult> execute(
+      IClientManager<PartitionRegionId, ConfigNodeClient> clientManager)
+      throws InterruptedException {
+    SettableFuture<ConfigTaskResult> future = SettableFuture.create();
+    TClusterNodeInfos clusterNodeInfos = new TClusterNodeInfos();
+
+    if (config.isClusterMode()) {
+      try (ConfigNodeClient client = clientManager.borrowClient(ConfigNodeInfo.partitionRegionId)) {
+        clusterNodeInfos = client.getAllClusterNodeInfos();
+      } catch (TException | IOException e) {
+        LOGGER.error("Failed to connect to config node.");
+        future.setException(e);
+      }
+    } else {
+      future.setException(new TException("SHOW CLUSTER not support local mode"));
+    }
+
+    // build TSBlock
+    TsBlockBuilder builder =
+        new TsBlockBuilder(HeaderConstant.showClusterHeader.getRespDataTypes());
+    int configNodeId = 0;
+    for (TConfigNodeLocation nodeLocation : clusterNodeInfos.getConfigNodeList()) {
+      builder.getTimeColumnBuilder().writeLong(0L);
+      builder.getColumnBuilder(0).writeInt(configNodeId++);
+      builder.getColumnBuilder(1).writeBinary(new Binary("ConfigNode"));
+      builder.getColumnBuilder(2).writeBinary(new Binary("Running"));
+      builder
+          .getColumnBuilder(3)
+          .writeBinary(new Binary(nodeLocation.getInternalEndPoint().getIp()));
+      builder.getColumnBuilder(4).writeInt(nodeLocation.getInternalEndPoint().getPort());
+      builder.declarePosition();
+    }
+    for (TDataNodeLocation nodeLocation : clusterNodeInfos.getDataNodeList()) {
+      builder.getTimeColumnBuilder().writeLong(0L);
+      builder.getColumnBuilder(0).writeInt(nodeLocation.getDataNodeId());
+      builder.getColumnBuilder(1).writeBinary(new Binary("DataNode"));
+      builder.getColumnBuilder(2).writeBinary(new Binary("Running"));
+      builder
+          .getColumnBuilder(3)
+          .writeBinary(new Binary(nodeLocation.getInternalEndPoint().getIp()));
+      builder.getColumnBuilder(4).writeInt(nodeLocation.getInternalEndPoint().getPort());
+      builder.declarePosition();
```

Comment:
Line `92-100` is similar with line `81-89`, suggest to define a function to process it.

Reply:
Line 92-100 and line 81-89 are extracted in a function